### PR TITLE
Allow unmounted resources past idle filters

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
Previously resources without idleKeys (Cluster or Provider ID) would derail the idle allocations process, this was not a problem when Cluster was the only possible key however with with Provider ID, allocations may be missing this value. Rather than ending the process these instances now trigger logs  and let the process continue. The only case of this happening so far is Unmounted PVs which do not contribute to any computational resources (CPU, GPU or RAM). However if it were to take place on a resource with computational resources, this would be indicative of an issue with Provider ID being distributed correctly.

I have added logging in the places where it makes sense to draw attention to these resources without being too repetitive and given the unmounted idleKey to items missing their own idleKey

### CODE REVIEW

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [inline pkg/kubecost/allocation.go:1018]: So, if we aren't continuing any longer, then what happens? How does this affect idle by cluster?

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [inline pkg/kubecost/allocation.go:968]: Should we only swallow this error if we're doing idle by node? Are we sure that we want to alter the old behavior to support the new behavior?

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [inline pkg/kubecost/allocation.go:1339]: Do we know that this will always be the case? Isn't it possible (e.g. in an error state) to be missing a provider ID and not because the allocation is unmounted?

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [inline pkg/kubecost/allocation.go:1345]: Why is it the case that we should return "unmounted" and an error in all cases where an allocation is without a cluster ID?

**Review by nikovacevic — COMMENTED** [review]
@AjayTripathy may feel differently, but I have some questions about this before we merge, even if it fixes some short-term trouble.

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [reply pkg/kubecost/allocation.go:1339]: Another way of phrasing this is: how can we be certain that everything missing a ProviderID (or a Cluster in the other case) is, in fact, an Unmounted PV?

File: pkg/kubecost/allocation.go
Comment (Sean-Holcomb) [reply pkg/kubecost/allocation.go:968]: I am ignoring these because they imply a node without a cluster or a node without a provider ID which from what I can tell does not exist

File: pkg/kubecost/allocation.go
Comment (Sean-Holcomb) [reply pkg/kubecost/allocation.go:1018]: So all of these continues actually came from the last PR on this [802](https://github.com/kubecost/cost-model/pull/802/files) and before they actually returned. The behavior from cluster id that you are talking about preserving is a blank allocations screen with an error in the network tab. I don't think that was the right behavior to begin with but it also stood in the way of this feature so I changed it.

File: pkg/kubecost/allocation.go
Comment (Sean-Holcomb) [reply pkg/kubecost/allocation.go:1339]: I am not married to using the unmountedSuffix here, I could just as well switch it back to returning "", it just seemed more accurate given the the only instance of missing idleKey I have seen

**Comment by AjayTripathy** [comment]
> @AjayTripathy may feel differently, but I have some questions about this before we merge, even if it fixes some short-term trouble.

No need to rush this. I'll live with unmounted disappearing from this view if we want to take the time to get this right.

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [reply pkg/kubecost/allocation.go:968]: Hm. I'm not sure I agree that returning an error and failing the query is wrong here. (If this implies a node without a cluster, and that shouldn't be possible, then I think the only correct behavior upon encountering an error here is to return it.) But I don't want to obstruct. Can we at least log the error before moving on?

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [reply pkg/kubecost/allocation.go:1018]: Okay, thanks for the context.

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [reply pkg/kubecost/allocation.go:1339]: If you don't mind, then let's switch back to `""`. Just because at this moment that is the case, doesn't mean it will always be the case, and I don't see another reason to believe that "no provider ID" should always imply "unmounted PV".

File: pkg/kubecost/allocation.go
Comment (nikovacevic) [reply pkg/kubecost/allocation.go:1345]: Same as above.

**Review by nikovacevic — APPROVED** [review]
I don't want to obstruct getting this fixed, so approving with the hope that we can at least remove the "unmounted" bit.

File: pkg/kubecost/allocation.go
Comment (Sean-Holcomb) [reply pkg/kubecost/allocation.go]: Switched back now

File: pkg/kubecost/allocation.go
Comment (Sean-Holcomb) [reply pkg/kubecost/allocation.go]: added back in since it will not interfere with the feature

**Pull request merged**
